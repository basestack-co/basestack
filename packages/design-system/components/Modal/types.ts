import type { ReactNode } from "react";
import type { ButtonProps } from "../Button";
export type Size = "small" | "medium" | "large" | "fullWidth";

export interface ModalProps {
  /**
   * Modal size
   */
  size?: Size;
  /**
   * Modal title
   */
  title: string;
  /**
   * onClose callback
   */
  onClose: () => void;
  /**
   * Footer buttons
   */
  buttons: Array<ButtonProps>;
  /**
   * If modal is open
   */
  isOpen: boolean;
  /**
   * Sets a min-height to the sheet modal
   */
  minHeight?: number;
  /**
   * Sets the modal to full height in mobile
   */
  expandMobile?: boolean;
  /**
   * A React Element
   */
  children: ReactNode;
  /**
   * Callback when the animation ends
   */
  onAnimationEnd?: () => void;
  /**
   * Enables or disables close modal on click outside main content
   */
  closeOnClickOutside?: boolean;
}
