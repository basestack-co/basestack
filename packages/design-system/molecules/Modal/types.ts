export type Size = "small" | "medium" | "large" | "fullWidth";

export type Button = {
  text: string;
  onClick: () => void;
};

export interface ModalProps {
  /**
   * Modal size
   */
  size?: Size;
  /**
   * Modal title
   */
  title: string;
  /**
   * onClose callback
   */
  onClose: () => void;
  /**
   * Footer buttons
   */
  buttons: Array<Button>;
  /**
   * If modal is open
   */
  isOpen: boolean;
  /**
   * Sets a min-height to the sheet modal
   */
  minHeight?: number;
  /**
   * Sets the modal to full height in mobile
   */
  expandMobile?: boolean;
}
