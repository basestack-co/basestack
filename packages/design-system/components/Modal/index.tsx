import { rem } from "polished";
import React, { memo } from "react";
import { animated, config, useTransition } from "react-spring";
import { useTheme } from "styled-components";
import { fadeIn } from "../../animations/springs";
import { Button, ButtonVariant } from "../Button";
import IconButton from "../IconButton";
import Text from "../Text";
import {
  Body,
  Container,
  ContentContainer,
  Footer,
  GlobalStyle,
  Header,
  Overlay,
  Sheet,
} from "./styles";
import { ModalProps } from "./types";

const AnimatedModal = animated(Container);
const AnimatedSheet = animated(Sheet);

const Modal: React.FC<ModalProps> = ({
  size = "small",
  isOpen = false,
  expandMobile = false,
  closeOnClickOutside = true,
  title,
  onClose,
  buttons,
  minHeight,
  children,
  onAnimationEnd,
}) => {
  const theme = useTheme();

  const transitionModal = useTransition(isOpen, {
    config: { ...config.stiff, duration: 300 },
    ...fadeIn,
    onRest: () => {
      if (!isOpen && typeof onAnimationEnd === "function") {
        onAnimationEnd();
      }
    },
  });

  const transitionSheet = useTransition(isOpen, {
    config: { ...config.stiff, duration: 300 },
    from: { opacity: 0, transform: `translate3d(0px, 50px, 0px)` },
    enter: { opacity: 1, transform: `translate3d(0px, 0px, 0px)` },
    leave: { opacity: 0, transform: `translate3d(0px, 50px, 0px)` },
  });

  return transitionModal(
    (styles, item) =>
      item && (
        <AnimatedModal
          expandMobile={expandMobile}
          style={styles}
          data-testid="modal-container"
        >
          <GlobalStyle />
          {transitionSheet(
            (styles, item) =>
              item && (
                <AnimatedSheet
                  size={size}
                  minHeight={minHeight ?? 0}
                  style={styles}
                >
                  <Header>
                    <Text size="xLarge" mr={theme.spacing.s2}>
                      {title}
                    </Text>
                    <IconButton ml="auto" onClick={onClose} icon="close" />
                  </Header>
                  <ContentContainer>
                    <Body
                      pb={buttons && buttons.length > 0 ? "2px" : rem("22px")}
                    >
                      {children}
                    </Body>
                    {buttons && buttons.length > 0 && (
                      <Footer expandMobile={expandMobile}>
                        {buttons.map((item, index, { length }) => {
                          const isLastItem = index + 1 === length;
                          return (
                            <Button
                              key={index.toString()}
                              variant={
                                item.variant
                                  ? item.variant
                                  : isLastItem
                                    ? ButtonVariant.Primary
                                    : ButtonVariant.Neutral
                              }
                              onClick={item.onClick}
                              isDisabled={item.isDisabled}
                              isLoading={item.isLoading}
                              ml={theme.spacing.s1}
                            >
                              {item.children}
                            </Button>
                          );
                        })}
                      </Footer>
                    )}
                  </ContentContainer>
                </AnimatedSheet>
              ),
          )}
          <Overlay {...(closeOnClickOutside ? { onClick: onClose } : {})} />
        </AnimatedModal>
      ),
  );
};

export { type ModalProps };

export default memo(Modal);
