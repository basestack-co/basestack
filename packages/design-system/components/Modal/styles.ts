import styled, { css, createGlobalStyle } from "styled-components";
import { transparentize, rem } from "polished";
import { Size } from "./types";

export const GlobalStyle = createGlobalStyle`
  body {
    overflow: hidden;
  }
`;

const flexColumn = css`
  display: flex;
  flex-direction: column;
`;

const flexRowCenter = css`
  display: flex;
  align-items: center;
`;

export const Container = styled.div<{ expandMobile: boolean }>`
  position: fixed;
  inset: 0;
  z-index: ${({ theme }) => theme.zIndex.modal};
  ${flexColumn};
  align-items: center;
  justify-content: center;
  padding: 0 ${({ theme }) => theme.spacing.s5};

  ${({ expandMobile }) =>
    expandMobile &&
    css`
      @media screen and ${({ theme }) => theme.device.max.sm} {
        align-items: initial;
        justify-content: flex-end;
        padding: 0;
      }
    `};
`;

const handleSheetSize = (size: Size) => {
  switch (size) {
    default:
    case "small":
      return "576px";
    case "medium":
      return "768px";
    case "large":
      return "992px";
  }
};

export const Sheet = styled.div<{
  size: Size;
  minHeight: number;
  expandMobile: boolean;
}>`
  max-width: ${({ size }) => handleSheetSize(size)};
  width: 100%;
  ${flexColumn};
  min-height: ${({ minHeight }) => `${minHeight}px`};
  background-color: ${({ theme }) => theme.modal.backgroundColor};
  box-shadow: ${({ theme }) => theme.shadow.elevation4};
  border-radius: 4px;
  z-index: 1;

  ${({ expandMobile }) =>
    expandMobile &&
    css`
      @media screen and ${({ theme }) => theme.device.max.sm} {
        max-height: calc(100vh - ${({ theme }) => theme.spacing.s6});
      }
    `};
`;

export const Header = styled.div`
  height: ${rem("75px")};
  padding: 0 ${rem("20px")};
  ${flexRowCenter};
`;

export const Body = styled.div`
  padding: 0 ${rem("20px")} ${rem("20px")} ${rem("20px")};
  ${flexColumn};
`;

export const Footer = styled.div`
  padding: ${rem("20px")};
  ${flexRowCenter};
  justify-content: flex-end;
  margin-top: auto;
`;

export const Overlay = styled.div`
  position: absolute;
  inset: 0;
  z-index: 0;
  background-color: ${({ theme }) => theme.modal.overlay.backgroundColor};
`;
