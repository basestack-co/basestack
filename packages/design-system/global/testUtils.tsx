import { type RenderOptions, render } from "@testing-library/react";
import type React from "react";
// Styles
import { ThemeProvider } from "styled-components";
import theme from "../theme/lightTheme";

interface Props {
  children: React.ReactNode;
}

const WithThemeProvider = ({ children }: Props) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithTheme = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: WithThemeProvider, ...options });

export { renderWithTheme };
