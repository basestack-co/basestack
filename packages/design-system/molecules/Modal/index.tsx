import React, { memo, useState, useEffect } from "react";
import { Button, ButtonVariant, IconButton, Text } from "../../atoms";
import {
  Body,
  Container,
  Footer,
  GlobalStyle,
  Header,
  Overlay,
  Sheet,
} from "./styles";
import { useTheme } from "styled-components";
import { animated, config, useTransition } from "react-spring";
import { ModalProps } from "./types";
import { fadeIn } from "../../animations/springs";

const AnimatedModal = animated(Container);
const AnimatedSheet = animated(Sheet);

const Modal: React.FC<ModalProps> = ({
  size = "small",
  isOpen = false,
  expandMobile = false,
  title,
  onClose,
  buttons,
  minHeight,
  children,
  onAnimationEnd,
}) => {
  const theme = useTheme();

  const transitionModal = useTransition(isOpen, {
    config: { ...config.stiff, duration: 300 },
    ...fadeIn,
    onRest: (result: any) => {
      if (!result && onAnimationEnd) {
        onAnimationEnd();
      }
    },
  });

  const transitionSheet = useTransition(isOpen, {
    config: { ...config.stiff, duration: 300 },
    from: { opacity: 0, transform: `translate3d(0px, 50px, 0px)` },
    enter: { opacity: 1, transform: `translate3d(0px, 0px, 0px)` },
    leave: { opacity: 0, transform: `translate3d(0px, 50px, 0px)` },
  });

  return transitionModal(
    (styles, item) =>
      item && (
        <AnimatedModal
          expandMobile={expandMobile}
          style={styles}
          data-testid="modal-container"
        >
          <GlobalStyle />
          {transitionSheet(
            (styles, item) =>
              item && (
                <AnimatedSheet
                  expandMobile={expandMobile}
                  size={size}
                  minHeight={minHeight}
                  style={styles}
                >
                  <Header>
                    <Text size="xLarge" mr={theme.spacing.s2}>
                      {title}
                    </Text>
                    <IconButton ml="auto" onClick={onClose} icon="close" />
                  </Header>
                  <Body>{children}</Body>
                  <Footer>
                    {buttons &&
                      buttons.map((item, index, { length }) => {
                        const isLastItem = index + 1 === length;
                        return (
                          <Button
                            key={index.toString()}
                            variant={
                              item.variant
                                ? item.variant
                                : isLastItem
                                ? ButtonVariant.Primary
                                : ButtonVariant.Neutral
                            }
                            onClick={item.onClick}
                            isDisabled={item.isDisabled}
                            isLoading={item.isLoading}
                            ml={theme.spacing.s1}
                          >
                            {item.children}
                          </Button>
                        );
                      })}
                  </Footer>
                </AnimatedSheet>
              )
          )}
          <Overlay onClick={onClose} />
        </AnimatedModal>
      )
  );
};

export default memo(Modal);
