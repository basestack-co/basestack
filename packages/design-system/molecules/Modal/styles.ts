import styled, { css, createGlobalStyle } from "styled-components";
import { transparentize, rem } from "polished";

export const GlobalStyle = createGlobalStyle`
  body {
    overflow: hidden;
  }
`;

const flexColumn = css`
  display: flex;
  flex-direction: column;
`;

const flexRowCenter = css`
  display: flex;
  align-items: center;
`;

export const Container = styled.div`
  position: fixed;
  inset: 0;
  z-index: ${({ theme }) => theme.zIndex.modal};
  ${flexColumn};
  align-items: center;
  justify-content: center;
`;

export const Sheet = styled.div`
  max-width: 500px;
  width: 100%;

  ${flexColumn};
  height: 300px;
  background-color: ${({ theme }) => theme.colors.white};
  box-shadow: ${({ theme }) => theme.shadow.elevation4};
  border-radius: 4px;
  z-index: 1;
`;

export const Header = styled.div`
  height: ${rem("75px")};
  padding: 0 ${rem("20px")};
  ${flexRowCenter};
`;

export const Body = styled.div`
  padding: 0 ${rem("20px")};
  ${flexColumn};
`;

export const Footer = styled.div`
  padding: ${rem("20px")};
  ${flexRowCenter};
  justify-content: flex-end;
  margin-top: auto;
`;

export const Overlay = styled.div`
  position: absolute;
  inset: 0;
  z-index: 0;
  background-color: ${({ theme }) => transparentize("0.5", theme.colors.black)};
`;
