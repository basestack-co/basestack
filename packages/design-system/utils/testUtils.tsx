import defaultTheme from "@basestack/design-system/theme/lightTheme";
import isPropValid from "@emotion/is-prop-valid";
import { RenderResult, render } from "@testing-library/react";
import React from "react";
import { StyleSheetManager, ThemeProvider } from "styled-components";

const Provider = ({
  children,
  theme = defaultTheme,
}: {
  children: React.ReactNode;
  theme: any;
}) => (
  <ThemeProvider theme={theme}>
    <StyleSheetManager shouldForwardProp={isPropValid}>
      {children}
    </StyleSheetManager>
  </ThemeProvider>
);

export const renderWithTheme = (
  ui: React.ReactElement,
  options?: any,
): RenderResult => render(ui, { wrapper: Provider, ...options });
