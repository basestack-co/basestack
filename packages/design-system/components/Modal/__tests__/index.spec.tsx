import { cleanup } from "@testing-library/react";
import { renderWithTheme } from "../../../utils/testUtils";
import Modal from "../index";

describe("Modal tests", () => {
  afterEach(cleanup);

  const buttons = [
    { children: "Close", onClick: jest.fn },
    { children: "Create", onClick: jest.fn },
  ];

  test("should render Modal correctly", () => {
    const { asFragment } = renderWithTheme(
      <Modal
        title="Create Flag"
        isOpen={false}
        onClose={jest.fn()}
        buttons={buttons}
      >
        Body Content
      </Modal>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Modal correctly open", () => {
    const { asFragment } = renderWithTheme(
      <Modal title="Create Flag" isOpen onClose={jest.fn()} buttons={buttons}>
        Body Content
      </Modal>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("should render Modal with custom props", () => {
    const { asFragment } = renderWithTheme(
      <Modal
        title="Create Flag Large"
        isOpen
        onClose={jest.fn()}
        buttons={buttons}
        size="large"
        minHeight={300}
      >
        Body Content
      </Modal>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
