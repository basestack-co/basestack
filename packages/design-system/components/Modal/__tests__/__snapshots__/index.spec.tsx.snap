// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests should render Modal correctly 1`] = `<DocumentFragment />`;

exports[`Modal tests should render Modal correctly open 1`] = `
<DocumentFragment>
  .c5 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c6 {
  color: #000000;
  font-size: 1.5rem;
}

.c10 {
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  width: normal;
  height: normal;
  margin-left: 0.25rem;
  width: fit-content;
}

.c10:hover:not(:disabled) {
  background-color: #EEEEEE;
}

.c10:active:not(:disabled) {
  background-color: #E2E2E2;
}

.c10 .material-symbols-sharp {
  color: inherit;
}

.c10 svg,
.c10 svg g {
  fill: #000000;
}

.c10:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c11 {
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #276EF1;
  color: #FFFFFF;
  width: normal;
  height: normal;
  margin-left: 0.25rem;
  width: fit-content;
}

.c11:hover:not(:active, :disabled) {
  background-color: #0e55d7;
}

.c11 .material-symbols-sharp {
  color: inherit;
}

.c11 svg,
.c11 svg g {
  fill: #FFFFFF;
}

.c11:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c4 {
  margin-left: auto;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2.25rem;
  width: 2.25rem;
}

.c4 .material-symbols-sharp {
  color: inherit;
}

.c4:hover {
  background-color: #EEEEEE;
}

.c4:active {
  background-color: #E2E2E2;
}

.c3 {
  font-size: 1.25rem;
  line-height: 1.875rem;
  font-weight: 500;
  font-family: roboto;
  margin-right: 0.5rem;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #000000;
}

.c0 {
  position: fixed;
  inset: 0;
  z-index: 110;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 1.25rem;
}

.c1 {
  display: flex;
  flex-direction: column;
  max-width: 576px;
  width: 100%;
  min-height: 0px;
  background-color: #FFFFFF;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 10px 15px -3px,rgba(0, 0, 0, 0.05) 0px 4px 6px -2px;
  border-radius: 4px;
  z-index: 1;
  max-height: calc(100vh - 1.875rem);
}

.c7 {
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.c7::-webkit-scrollbar {
  width: 4px;
}

.c7::-webkit-scrollbar-track {
  background: #EEEEEE;
}

.c7::-webkit-scrollbar-thumb {
  background: #CBCBCB;
}

.c7::-webkit-scrollbar-thumb:hover {
  background: #AFAFAF;
}

.c2 {
  height: 4.75rem;
  padding: 0 1.25rem;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  display: flex;
  align-items: center;
  background-color: #FFFFFF;
  flex-shrink: 0;
}

.c8 {
  padding: 2px 1.25rem;
  padding-bottom: 2px;
  display: flex;
  flex-direction: column;
}

.c9 {
  background-color: #FFFFFF;
  height: 4.75rem;
  padding: 0 1.25rem;
  display: flex;
  align-items: center;
  flex-shrink: 0;
  justify-content: flex-end;
  margin-top: auto;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c12 {
  position: absolute;
  inset: 0;
  z-index: 0;
  background-color: rgba(0,0,0,0.5);
}

<div
    class="c0"
    data-testid="modal-container"
    style="opacity: 0;"
  >
    <div
      class="c1"
      style="opacity: 0; transform: translate3d(0px, 50px, 0px);"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
          font-family="roboto"
        >
          Create Flag
        </h2>
        <button
          class="c4"
        >
          <div
            class="c5"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c6 material-symbols-rounded"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              close
            </span>
          </div>
        </button>
      </div>
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          Body Content
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
            data-testid="button"
          >
            <span
              class=""
            >
              Close
            </span>
          </button>
          <button
            class="c11"
            data-testid="button"
          >
            <span
              class=""
            >
              Create
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="c12"
    />
  </div>
</DocumentFragment>
`;

exports[`Modal tests should render Modal with custom props 1`] = `
<DocumentFragment>
  .c5 {
  height: 1.5rem;
  width: 1.5rem;
  flex-shrink: 0;
}

.c6 {
  color: #000000;
  font-size: 1.5rem;
}

.c10 {
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  width: normal;
  height: normal;
  margin-left: 0.25rem;
  width: fit-content;
}

.c10:hover:not(:disabled) {
  background-color: #EEEEEE;
}

.c10:active:not(:disabled) {
  background-color: #E2E2E2;
}

.c10 .material-symbols-sharp {
  color: inherit;
}

.c10 svg,
.c10 svg g {
  fill: #000000;
}

.c10:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c11 {
  height: 2.25rem;
  padding: 0 0.75rem;
  font-size: 0.875rem;
  position: relative;
  border: none;
  display: flex;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
  border-radius: 4px;
  text-decoration: none;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: #276EF1;
  color: #FFFFFF;
  width: normal;
  height: normal;
  margin-left: 0.25rem;
  width: fit-content;
}

.c11:hover:not(:active, :disabled) {
  background-color: #0e55d7;
}

.c11 .material-symbols-sharp {
  color: inherit;
}

.c11 svg,
.c11 svg g {
  fill: #FFFFFF;
}

.c11:disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c4 {
  margin-left: auto;
  border: none;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  border-radius: 50%;
  transition: background-color 0.1s ease-in-out,color 0.1s ease-in-out,border-color 0.1s ease-in-out;
  background-color: transparent;
  color: #000000;
  height: 2.25rem;
  width: 2.25rem;
}

.c4 .material-symbols-sharp {
  color: inherit;
}

.c4:hover {
  background-color: #EEEEEE;
}

.c4:active {
  background-color: #E2E2E2;
}

.c3 {
  font-size: 1.25rem;
  line-height: 1.875rem;
  font-weight: 500;
  font-family: roboto;
  margin-right: 0.5rem;
  font-family: 'Roboto',sans-serif;
  word-break: break-word;
  color: #000000;
}

.c0 {
  position: fixed;
  inset: 0;
  z-index: 110;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 1.25rem;
}

.c1 {
  display: flex;
  flex-direction: column;
  max-width: 992px;
  width: 100%;
  min-height: 300px;
  background-color: #FFFFFF;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 10px 15px -3px,rgba(0, 0, 0, 0.05) 0px 4px 6px -2px;
  border-radius: 4px;
  z-index: 1;
  max-height: calc(100vh - 1.875rem);
}

.c7 {
  display: flex;
  flex-direction: column;
  overflow: auto;
}

.c7::-webkit-scrollbar {
  width: 4px;
}

.c7::-webkit-scrollbar-track {
  background: #EEEEEE;
}

.c7::-webkit-scrollbar-thumb {
  background: #CBCBCB;
}

.c7::-webkit-scrollbar-thumb:hover {
  background: #AFAFAF;
}

.c2 {
  height: 4.75rem;
  padding: 0 1.25rem;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  display: flex;
  align-items: center;
  background-color: #FFFFFF;
  flex-shrink: 0;
}

.c8 {
  padding: 2px 1.25rem;
  padding-bottom: 2px;
  display: flex;
  flex-direction: column;
}

.c9 {
  background-color: #FFFFFF;
  height: 4.75rem;
  padding: 0 1.25rem;
  display: flex;
  align-items: center;
  flex-shrink: 0;
  justify-content: flex-end;
  margin-top: auto;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c12 {
  position: absolute;
  inset: 0;
  z-index: 0;
  background-color: rgba(0,0,0,0.5);
}

<div
    class="c0"
    data-testid="modal-container"
    style="opacity: 0;"
  >
    <div
      class="c1"
      style="opacity: 0; transform: translate3d(0px, 50px, 0px);"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
          font-family="roboto"
        >
          Create Flag Large
        </h2>
        <button
          class="c4"
        >
          <div
            class="c5"
            data-testid="icon-container"
            size="24"
          >
            <span
              class="c6 material-symbols-rounded"
              color="#000000"
              data-testid="icon"
              size="24"
            >
              close
            </span>
          </div>
        </button>
      </div>
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          Body Content
        </div>
        <div
          class="c9"
        >
          <button
            class="c10"
            data-testid="button"
          >
            <span
              class=""
            >
              Close
            </span>
          </button>
          <button
            class="c11"
            data-testid="button"
          >
            <span
              class=""
            >
              Create
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="c12"
    />
  </div>
</DocumentFragment>
`;
